document/src/vespa/document/base/idstring.cpp

```diff
@@ -153,8 +153,13 @@ inline const char *
 fmemchr(const char * s, const char * e)
 {
     while (s+15 < e) {
+#ifdef __clang__
+        v16qi tmpCurrent = __builtin_ia32_lddqu(s);
```

Comment:
Finnes det ikke noe som fungerer for begge ?

Reply:
Something like this _should_ be portable (warning: I have not verified 100% identical semantics):
```c++
#include <emmintrin.h>
(...)
inline const char *
fmemchr(const char * s, const char * e)
{
    const auto sep = _mm_set1_epi8(':');
    while (s+15 < e) {
        auto current = _mm_loadu_si128(reinterpret_cast<const __m128i*>(s));
        auto mask = _mm_movemask_epi8(_mm_cmpeq_epi8(current, sep));
        if (__builtin_expect(mask, 1)) {
            return s + lsbIdx(uint32_t(mask));
        }
        s+=16;
    }
(...)
```

Reply:
Both GCC and clang seems to happily eat the above for dinner, at least: https://godbolt.org/z/08UbwY